File: benchmarks/fib.rs
// fib.rs — Rust reference implementation for the Arukellt fib benchmark.
// Matches benchmarks/fib.ark: iterative Fibonacci(35).
// Compile: rustc -O -o fib fib.rs
// Run:     ./fib

fn fib(n: i32) -> i32 {
    if n <= 1 {
        return n;
    }
    let (mut a, mut b) = (0i32, 1i32);
    for _ in 2..=n {
        let next = a + b;
        a = b;
        b = next;
    }
    b
}

fn main() {
    println!("{}", fib(35));
}

File: benchmarks/legacy/fib.c
/* fib.c — C reference implementation for the Arukellt fib benchmark.
 * Matches benchmarks/fib.ark: iterative Fibonacci(35).
 * Compile: cc -O2 -o fib fib.c
 * Run:     ./fib
 */
#include <stdio.h>

static int fib(int n) {
    if (n <= 1)
        return n;
    int a = 0, b = 1;
    for (int i = 2; i <= n; i++) {
        int next = a + b;
        a = b;
        b = next;
    }
    return b;
}

int main(void) {
    printf("%d\n", fib(35));
    return 0;
}
